### PR TITLE
Add notebook testing capabilites in Interactive

### PR DESCRIPTION
**Pull request by rahulbshrestha**
Related PR: https://github.com/scverse/spatialdata-notebooks/pull/46

### CODE REVIEW

File: src/napari_spatialdata/_interactive.py [hunk 1]
```diff
@@ -51,7 +55,32 @@ def add_element(self, coordinate_system_name: str, element: str) -> SpatialData:
...
-    def __init__(self, sdata: SpatialData, headless: bool = False) -> None:
+    def create_folders(self, tested_notebook, test_target):
+        main_folder = os.getcwd()  # Get the current working directory
+        tests_folder = os.path.join(main_folder, "tests")
+        notebook_folder = os.path.join(tests_folder, tested_notebook)
+        cell_folder = os.path.join(notebook_folder, test_target)
+
+        print("Creating folders: !!!")
+        if not os.path.exists(tests_folder):
+            os.mkdir(tests_folder)
```
Comment (LucaMarconato) [inline src/napari_spatialdata/_interactive.py:66]: please remove the if and use os.makedirs(path, exists_ok=True)

File: src/napari_spatialdata/_interactive.py [hunk 2]
```diff
@@ -60,11 +89,27 @@ def __init__(self, sdata: SpatialData, headless: bool = False) -> None:
             self._sdata_widget, name="SpatialData", area="left", menu=self._viewer.window.window_menu
         )
         self._viewer.window.add_plugin_dock_widget("napari-spatialdata", "View")
+
         if not headless:
             self.run()
 
+        if tested_notebook is not None:
+            filepath = self.create_folders(tested_notebook, test_target)
+            fig, axs = plt.subplots(ncols=3, figsize=(12, 3))
```
Comment (LucaMarconato) [inline src/napari_spatialdata/_interactive.py:98]: why ncols=3? what is this?

File: src/napari_spatialdata/_interactive.py [hunk 3]
```diff
@@ -60,11 +89,27 @@ def __init__(self, sdata: SpatialData, headless: bool = False) -> None:
...
+
+            for element_type, element_name, _ in self._sdata.filter_by_coordinate_system("global")._gen_elements():
+                self.add_element(coordinate_system_name="global", element=element_name)
+
+                axis_count = 0
+
+                if take_screenshot:
+                    save_image(self.screenshot(canvas_only=True), filepath + "/" + element_name + ".png")
```
Comment (LucaMarconato) [inline src/napari_spatialdata/_interactive.py:106]: please use os.path.join() or pathlib.Path() `/` operator, it's more robust.

File: src/napari_spatialdata/_interactive.py [hunk 4]
```diff
@@ -60,11 +89,27 @@ def __init__(self, sdata: SpatialData, headless: bool = False) -> None:
...
         self._viewer.window.add_plugin_dock_widget("napari-spatialdata", "View")
+
         if not headless:
             self.run()
 
+        if tested_notebook is not None:
+            filepath = self.create_folders(tested_notebook, test_target)
+            fig, axs = plt.subplots(ncols=3, figsize=(12, 3))
+
+            for element_type, element_name, _ in self._sdata.filter_by_coordinate_system("global")._gen_elements():
```
Comment (LucaMarconato) [inline src/napari_spatialdata/_interactive.py:100]: As suggested by @melonora in the other PR, the coordinate system could be passed to interactive. Please add an optional parameter, default to None, for doing this.

File: src/napari_spatialdata/_interactive.py [hunk 5]
```diff
@@ -60,11 +89,27 @@ def __init__(self, sdata: SpatialData, headless: bool = False) -> None:
             self._sdata_widget, name="SpatialData", area="left", menu=self._viewer.window.window_menu
         )
         self._viewer.window.add_plugin_dock_widget("napari-spatialdata", "View")
+
         if not headless:
```
Comment (LucaMarconato) [inline src/napari_spatialdata/_interactive.py:93]: This if should be added last, otherwise the interface may get blocked.

**Comment by LucaMarconato** [comment]
I merged the other PR, thanks for the contribution. Please pull from main before continuing.

**Comment by giovp** [comment]
I think this can actually be closed

**Comment by melonora** [comment]
ok, I would say it would be good to add tests for notebooks though, but low priority for now.

**Pull request closed without merging**